docs/api_proposal.md

```diff
@@ -0,0 +1,146 @@
+
+## Methods
+
+Starting with GET methods. At a high level, we need methods to
+1. Search for taxa based on name strings (taxonomic name resolution)
+1. Search taxa based on some parameter
+1. Get a specific  taxon
+1. Return information about the taxonomic hierarchy / relationships between taxa.
+
+Here is a proposed list of key GET methods:
+
+### /tnrs
+Method for taxonomic name reconciliation; returns potentially matching taxa (exact or fuzzy matching; uses information about synonyms)
+
+`/tnrs?name="Genus speices"`
+
+returns:
+```
+{
+  "numberMatches" : 3,
+  "bestMatch" : {
+    "id" : 3,
+    "canonicalName" : "Genus species",
+    "matchScore" : 0.8
+    },
+  "alternateMatches" : [
+    { _info about next match here_ },
+    { _info about next match here_ }
+  ]
+}
+```
+
+### /autocomplete
+Return a list of potentially matching taxa based on given prefix; for UI search boxes
+
+`/autocomplete?prefix="Gen"`
+
+Return format same as [tnrs](#tnrs).
+
+### /search
+Search taxa based on a variety of parameters (rank, author, year, etc)  
+
+`/search?author="Linnaeus"&rank="order"`
+
+Return format same as [tnrs](#tnrs).
+
+### /taxon
+Return a specific taxon resource; takes ID as parameter
+
+  `/taxon/{id}`
+
+returns:
+```
+{
+  "id" : id,
+  "canonicalName" : "Genus species",
+  "scientificName" : "Genus species (author, year)",
+  "synonyms" : ["synonym1", "synonymn2"],
+  "rank" : "genus"
+}
+```
+
+### /about
+Return information about a taxonomic hierarchy.
+
+```
+{
+  "taxonomyId" : id,
+  "lastEdited" : "2018-12-14",
+  "numberTaxa" : 96323,
+  "rootTaxonId" : 2,
+  "rootTaxonName" : "Life"
+}
+```
+
+### /mrca
+
+Return the most recent common ancestor of two or more taxa
+
+  `/mrca?taxa=["Genus sp1", "Genus sp2"]`
+
+returns:
+```
+{
+  "mrca" : {
+    "id" : id,
+    "canonicalName" : "Genus species",
+    "scientificName" : "Genus species (author, year)",
+    "synonyms" : ["synonym1", "synonymn2"],
+    "rank" : "genus"
+  }
+}
+```
+
+### /subtree
+Return the subtree descended from a taxon
+
+`/subtree?taxon=id`
+
+Returns json with subtree, limited and / or paginated if large number of levels / children.
+
+## Questions
+
+* Will a single DINA system store more than one taxonomy?
+  * if more than one, how does a API user disambiguate?
+  * disambiguation required for all methods?
+* What is the minimum metadata required to create a new taxon?
+* IDs vs names
+  * REST APIs generally operate on a resource (a unique thing) and names are not unique (see Assumptions)
+  * workflow proposed here: user provides a name, we give back a list of potentially matching names. Each returned name has some metadata (enough for a user to decide which one they want) and an identifier. Using the identifier, they can access operations on that taxon.
+  * which methods only take IDs vs names?
+* assigning IDs to taxa
+  * what entity does an ID apply to?
```

Comment:
this is directly in link with https://github.com/DINA-Web/taxonomy/blob/master/docs/taxonomy_concept_diagram.png